ambry-messageformat/src/main/java/com.github.ambry.messageformat/TtlUpdateMessageFormatInputStream.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.messageformat;
+
+import com.github.ambry.store.StoreKey;
+import com.github.ambry.utils.Utils;
+import java.nio.ByteBuffer;
+
+
+/**
+ * Represents a TTL update message.
+ *
+ *  - - - - - - - - - - - - -
+ * |     Message Header      |
+ *  - - - - - - - - - - - - -
+ * |       blob key          |
+ *  - - - - - - - - - - - - -
+ * |     update Record       |
+ *  - - - - - - - - - - - - -
+ *
+ */
+public class TtlUpdateMessageFormatInputStream extends MessageFormatInputStream {
```

Comment:
This is awfully similar to what `DeleteMessageFormatInputStream` does. Fundamentally, since these are both the same record (with different sub-records), do you think it is cleaner and less verbose if you instead had a common `UpdateMessageFormatInputStream` that took care of both?

Reply:
They look similar but I think the fact that subrecord is different means that their constructors may take in different arguments (in the future) because they are different kinds of updates. For e.g., if we support TTL update such that an expiry time can be specified (instead of just permanent), then `TtlUpdateMessageFormatInputStream` would receive a "updatedExpiryTimeMs" which would not be applicable to the delete. Do you still think they should be unified?

Reply:
Since the code is mostly common and there is a logical relationship between the two, I think an option is to have most of the common code within a private constructor, and expose public constructors where things are different (just like you do with the `UpdateRecord`). What do you think?

Reply:
Tried this out - it is certainly possible but not at current time because `DeleteMessageFormatInputStream` is on update format V2 but `TtlUpdateMessageFormatInputStream` is on update format V3. When I switch `DeleteMessageFormatInputStream` to V3, I will also combine the code.